src/borg/archive.py

```diff
@@ -468,6 +469,8 @@ def save(self, name=None, comment=None, timestamp=None, additional_metadata=None
         data = self.key.pack_and_authenticate_metadata(metadata.as_dict(), context=b'archive')
         self.id = self.key.id_hash(data)
         self.cache.add_chunk(self.id, Chunk(data), self.stats)
+        while self.repository.async_response(wait=True) is not None:
```

Comment:
as there is nothing except exceptions yet in the responses, this will either terminate or raise immediately.